pkg/operator/prunecontroller/prune_controller.go

```diff
@@ -96,7 +100,32 @@ func (c *PruneController) sync() error {
 
 	for _, s := range secretsToBePruned(minRevision, c.secretPrefixes, secrets) {
 		klog.V(4).Infof("Pruning old secret %q", s.Name)
-		if err := c.secretGetter.Secrets(s.Namespace).Delete(s.Name, nil); err != nil {
+
+		// remove finalizer
+		retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			s, err := c.secretGetter.Secrets(s.Namespace).Get(s.Name, metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					return nil
+				}
+				return err
+			}
+
+			// remove finalizer
+			newFinalizers := make([]string, 0, len(s.Finalizers))
+			for _, f := range s.Finalizers {
+				if f == encryptionsecrets.EncryptionSecretFinalizer {
+					continue
+				}
+				newFinalizers = append(newFinalizers, f)
+			}
+			s.Finalizers = newFinalizers
+
+			_, err = c.secretGetter.Secrets(s.Namespace).Update(s)
+			return err
+		})
+
+		if err := c.secretGetter.Secrets(s.Namespace).Delete(s.Name, nil); err != nil && !errors.IsNotFound(err) {
```

Comment:
ok, so regardless of the status of the previous operation (success, failure) remove the secret, why not :)

Reply:
Worth a try.